src/TestFramework/TestFramework/Assertions/Assert.AreSame.cs

```diff
@@ -30,10 +31,8 @@ public sealed partial class Assert
     /// Thrown if <paramref name="expected"/> does not refer to the same object
     /// as <paramref name="actual"/>.
     /// </exception>
-    public static void AreSame<T>(T? expected, T? actual)
-    {
-        AreSame(expected, actual, string.Empty, null);
-    }
+    public static void AreSame<T>([NotNullIfNotNull(nameof(actual))] T? expected, [NotNullIfNotNull(nameof(expected))] T? actual)
```

Comment:
`AreSame` is like "are references equal" so it's ok to say that if one parameter is not null the other will always be non-null.